cli/cmd/install-service-mirror.go

```diff
@@ -0,0 +1,133 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/linkerd/linkerd2/pkg/charts"
+	"github.com/linkerd/linkerd2/pkg/charts/servicemirror"
+	"github.com/linkerd/linkerd2/pkg/version"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"k8s.io/helm/pkg/chartutil"
+	"sigs.k8s.io/yaml"
+)
+
+type installServiceMirrorOptions struct {
+	namespace    string
+	logLevel     string
+	image        string
+	version      string
+	uid          int64
+	requeueLimit int32
+}
+
+const helmServiceMirrorDefaultChartName = "linkerd2-service-mirror"
+
+func newCmdInstallServiceMirror() *cobra.Command {
+	options, err := newInstallServiceMirrorOptionsWithDefaults()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%s", err)
+		os.Exit(1)
+	}
+
+	cmd := &cobra.Command{
+		Use:   "install-service-mirror [flags]",
+		Short: "Output Kubernetes configs to install Linkerd Service Mirror",
+		Long:  "Output Kubernetes configs to install Linkerd Service Mirror",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return renderServiceMirror(os.Stdout, options)
+		},
+	}
+
+	cmd.PersistentFlags().StringVarP(&options.namespace, "namespace", "n", options.namespace, "The namespace in which the Service Mirror component shall be installed")
+	cmd.PersistentFlags().StringVarP(&options.version, "version", "", options.version, "The Version of the Service Mirror component")
```

Comment:
We can, I was under the impression we want to make it a bit more stand alone. But yes I am all for that.

Reply:
My personal preference is that it is a separate binary, with its own chart, that is part of the addon flow @Pothulapati is working on right now. Obviously, it doesn't need to be that way in the beginning, but that feels like the correct end state. I can see a lot of folks installing and then deciding to add multi-cluster support at some later time, requiring a full `upgrade` feels weird.

Reply:
Just to be explicit here, there are two components:

* The service mirror binary itself.  Currently this is built into the controller binary, but it could be split out in the future.
* The Linkerd CLI command to emit the service mirror manifest.  That's what this code refers to.  Installing the service mirror does not require `linkerd install` or `linkerd upgrade` at all.  It is simply a matter of running `linkerd install-service-mirror | kubectl apply -f -`